sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/cli/ColumnValue.java

```diff
@@ -195,7 +195,7 @@ public static TColumnValue toTColumnValue(Type type, Object value) {
     case DECIMAL_TYPE:
       return stringValue(((HiveDecimal)value));
     case BINARY_TYPE:
-      return stringValue((String)value);
+      return stringValue(new String((byte[])value));
```

Comment:
String roundtrip with system default encoding (say, UTF-8) could be broken. Can we add a test? If that works, then should be good to go.

Reply:
How about: ```stringValue(UTF8String.fromBytes((byte[])value).toString());```?
https://github.com/apache/spark/blob/b367b323d2d69c6fcb8d4c83d87813e3924b7dc8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L245